user/caribou/module/src/AD9249EventConverter.cc

```diff
@@ -65,20 +75,42 @@ void AD9249Event2StdEventConverter::decodeChannel(
   timestamp = static_cast<uint64_t>(timestamp * 1e6 / 65.);
 }
 
-bool AD9249Event2StdEventConverter::Converting(
-    eudaq::EventSPC d1, eudaq::StandardEventSP d2,
-    eudaq::ConfigurationSPC conf) const {
+bool
+AD9249Event2StdEventConverter::Converting(eudaq::EventSPC d1,
+                                          eudaq::StandardEventSP d2,
+                                          eudaq::ConfigurationSPC conf) const {
 
-  if( !m_configured ){
+  if (!m_configured) {
     threshold_trig = conf->Get("threshold_trig", 1000);
     threshold_low = conf->Get("threshold_low", 101);
+    m_blStart = conf->Get("blStart", 200);
+    m_blEnd = conf->Get("blEnd", 100);
+    m_calib_range_min = conf->Get("calib_range_min", m_calib_range_min);
+    m_calib_range_max = conf->Get("calib_range_max", m_calib_range_max);
     m_useTime = conf->Get("use_time_stamp", false);
     m_waveform_filename = conf->Get("waveform_filename", "");
 
-    EUDAQ_DEBUG( "Using configuration:" );
-    EUDAQ_DEBUG( " threshold_low  = " + to_string( threshold_low ));
-    EUDAQ_DEBUG( " threshold_trig  = " + to_string( threshold_trig ));
-    EUDAQ_DEBUG( " use_time_stamp  = " + to_string( m_useTime ));
+    // read calibration functions
+    m_calib_functions.clear();
+    for (unsigned int i = 0; i < m_calib_strings.size(); i++) {
+      std::string name = "calibration_px" + to_string(mapping.at(i).first) +
+                         to_string(mapping.at(i).second);
+      m_calib_strings.at(i) = conf->Get(name, m_calib_strings.at(i));
+      m_calib_functions.emplace_back(name.c_str(), m_calib_strings.at(i).c_str(),
+                                     m_calib_range_min, m_calib_range_max);
+    }
+
+    EUDAQ_DEBUG("Using configuration:");
+    EUDAQ_DEBUG(" threshold_low  = " + to_string(threshold_low));
+    EUDAQ_DEBUG(" threshold_trig  = " + to_string(threshold_trig));
+    EUDAQ_DEBUG(" use_time_stamp  = " + to_string(m_useTime));
+    EUDAQ_DEBUG("Calibration functions: ");
+    for (unsigned int i = 0; i < m_calib_strings.size(); i++) {
```

Comment:
Does EUDAQ tell you somehow in which debug level it is? Then you woudn't have to do 16 loop iterations when nothing if printed (Corry has e.g. `IFLOG(DEBUG) { /*...*/ }`)

Reply:
It does `EUDAQ_IS_LOGGED`.